### PR TITLE
Feature join hangout

### PR DESCRIPTION
**Pull request by manuelrurda**
Users can now join hangouts using the hangout objectId as a code to find the hangout object they want to join.

### CODE REVIEW

File: app/src/main/java/com/example/hangouts/homeScreen/hangoutJoining/JoinHangoutViewModel.java [hunk 1]
```diff
@@ -22,22 +21,52 @@ public class JoinHangoutViewModel extends ViewModel {
...
+                        addHangoutToUser(hangout);
+                    }
+                }
+            });
+        } else {
+            errors.postValue(Errors.ERROR_USER_ALREADY_IN_HANGOUT);
         }
     }
 
+    private void addHangoutToUser(Hangout hangout) {
```
Comment (manuelrurda) [inline app/src/main/java/com/example/hangouts/homeScreen/hangoutJoining/JoinHangoutViewModel.java:46]: There might be room for improvement here. When the joining operation is made, the database needs two updates, one for the user object and one for the hangout object. But what would happen if for some reason one is successful and one fails. There would be inconsistencies in the database. I was thinking making a new table of a UserHangout relation so just one table needs updates when joining.

File: app/src/main/java/com/example/hangouts/homeScreen/hangoutJoining/JoinHangoutFragment.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            case ERROR_USER_ALREADY_IN_HANGOUT:
+                Toast.makeText(getContext(), "USER ALREADY IN HANGOUT", Toast.LENGTH_SHORT)
+                        .show();
+            break;
+            case ERROR_UPDATING_HANGOUT:
+                Toast.makeText(getContext(), "ERROR JOINING HANGOUT", Toast.LENGTH_SHORT)
+                        .show();
+            break;
+            case ERROR_ADDING_HANGOUT_TO_USER:
+                Toast.makeText(getContext(), "ERROR UPDATING USER", Toast.LENGTH_SHORT)
```
Comment (patjackson52) [inline app/src/main/java/com/example/hangouts/homeScreen/hangoutJoining/JoinHangoutFragment.java:80]: These strings should go in strings.xml

File: app/src/main/java/com/example/hangouts/homeScreen/hangoutJoining/JoinHangoutViewModel.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            });
+        }
+    }
+
+    enum Errors {
+        ERROR_HANGOUT_NOT_FOUND,
+        ERROR_EMPTY_HANGOUTID_STRING,
+        ERROR_USER_ALREADY_IN_HANGOUT,
+        ERROR_UPDATING_HANGOUT,
+        ERROR_ADDING_HANGOUT_TO_USER
```
Comment (patjackson52) [inline app/src/main/java/com/example/hangouts/homeScreen/hangoutJoining/JoinHangoutViewModel.java:102]: A nice improvement would be adding an @StringRes int to this Errors enum and declaring the error string resid here.  Then you can remove the switch statement above completely and just do Toast.makeToast(resources.getString(errorEnum.getErrorStringResId()))

**Review by patjackson52 — APPROVED** [review]
Couple of suggestions, looks good after that.

**Pull request merged**